kinbot/reaction_generator.py

```diff
@@ -195,51 +195,46 @@ def generate(self):
                     products=[] 
                     for st_pt in obj.products_final:
                         products.append(st_pt.chemid) 
-                    products.append(' ') 
-                    products.append(' ') 
-                    products.append(' ') 
+                    products.extend([' ', ' ', ' ']) 
                     barrier = (self.qc.get_qc_energy(instance_name)[1] - sp_energy) * constants.AUtoKCAL
-                    logging.info('\tReaction {} has a barrier of {} and lead to products {} {} {}'.format(instance_name,barrier,products[0],products[1],products[2]))
+                    logging.info('\tReaction {} has a barrier of {} and leads to products {} {} {}'.format(instance_name, barrier, products[0], products[1], products[2]))
                      
                     for i, st_pt in enumerate(obj.products_final):
                         chemid = st_pt.chemid
-                        orig_geom = copy.deepcopy(st_pt.geom)
                         e, st_pt.geom = self.qc.get_qc_geom(str(st_pt.chemid) + '_well', st_pt.natom)
                         if e < 0:
-                            logging.info('\tProduct optimization failed for {}, product {}'.format(instance_name,st_pt.chemid))
+                            logging.info('\tProduct optimization failed for {}, product {}'.format(instance_name, st_pt.chemid))
                             self.species.reac_ts_done[index] = -999
                             err = -1
                         elif e != 0:
                             err = -1
                         else:
                             e2, st_pt.energy = self.qc.get_qc_energy(str(st_pt.chemid) + '_well')
                             e2, st_pt.zpe = self.qc.get_qc_zpe(str(st_pt.chemid) + '_well')
-                            st_pt.bond_mx()
                             st_pt.characterize(0)  # not allowed to use the dimer option here
-                            st_pt.calc_chemid()
```

Comment:
Do we need to add back in the st_pt.bond_mx() and st_pt.calc_chemid() suctions, or have they become part of st_pt.characterize()?

Reply:
the characterize fn starts with `self.bond_mx()`, which was always like that. Also, the `find_conf_dihedral` calls `find_dihedral` which then calls the `calc_chemid`. I run my tests like this and it worked.